### PR TITLE
[Locales]locales_gossip_menu_option

### PR DESCRIPTION
**Pull request by Devalenok**
**Changes proposed:**

-  locales_gossip_menu_option

**Target branch(es):** master

**Issues addressed:** Closes #https://github.com/azerothcore/azerothcore-wotlk/issues/456#issuecomment-292496367

**Tests performed:** Build, tested in-game.

### CODE REVIEW

**Comment by Viste** [comment]
plaese add info Target branch(es): master and etc

**Comment by Viste** [comment]
thx

**Comment by ShinDarth** [comment]
please sync your branch with azerothcore master. Also a tip for the next time: do not open a PR from your fork's master branch, create another branch and keep your fork/master clean so everything will be easier for you

File: src/game/World/World.cpp [hunk 1]
```diff
@@ -1235,8 +1235,10 @@ void World::LoadConfigSettings(bool reload)
 
     // MySQL ping time interval
     m_int_configs[CONFIG_DB_PING_INTERVAL] = sConfigMgr->GetIntDefault("MaxPingTime", 30);
-
-    // misc
+  
+    m_bool_configs[CONFIG_LFG_LOCATION_ALL] = sConfigMgr->GetBoolDefault("LFG.Location.All", true); 
+    
+	// misc
```
Comment (ShinDarth) [inline src/game/World/World.cpp]: please fix indentation

File: src/game/World/World.h [hunk 2]
```diff
@@ -145,6 +145,7 @@ enum WorldBoolConfigs
     CONFIG_CHATLOG_BGROUND,
     CONFIG_AUTOBROADCAST,
     CONFIG_ALLOW_TICKETS,
+	CONFIG_LFG_LOCATION_ALL,
```
Comment (ShinDarth) [inline src/game/World/World.h]: please fix indentation

File: src/scripts/Custom/DoubleXP.cpp [hunk 3]
```diff
@@ -19,8 +19,8 @@ if(sConfigMgr->GetBoolDefault("DoubleXP.Enable", true))
         day == boost::date_time::Saturday ||
         day == boost::date_time::Sunday) {
 
-        amount = amount*15; 
-        ChatHandler(player->GetSession()).PSendSysMessage("Double XP is going on now!");}
+        amount = amount*25; 
+        ChatHandler(player->GetSession()).PSendSysMessage("Двойной XP происходит сейчас!");}
```
Comment (ShinDarth) [inline src/scripts/Custom/DoubleXP.cpp]: ???

File: src/scripts/Custom/DoubleXP.cpp [hunk 4]
```diff
@@ -34,7 +34,7 @@ if(sConfigMgr->GetBoolDefault("DoubleXP.Enable", true))
         day == boost::date_time::Saturday ||
         day == boost::date_time::Sunday) {
 
-        ChatHandler(player->GetSession()).PSendSysMessage("Double XP is going on now!");
+        ChatHandler(player->GetSession()).PSendSysMessage("Двойной XP происходит сейчас!");
```
Comment (ShinDarth) [inline src/scripts/Custom/DoubleXP.cpp]: ???

File: src/game/Battlegrounds/BattlegroundMgr.cpp [hunk 5]
```diff
@@ -40,7 +40,7 @@
 /*********************************************************/
 
 BattlegroundMgr::BattlegroundMgr() : m_ArenaTesting(false), m_Testing(false), 
-    m_lastClientVisibleInstanceId(0), m_NextAutoDistributionTime(0), m_NextPeriodicQueueUpdateTime(5*IN_MILLISECONDS), randomBgDifficultyEntry(999, 0, 80, 80, 0)
+		m_lastClientVisibleInstanceId(0), m_NextAutoDistributionTime(0), m_AutoDistributionTimeChecker(0), m_NextPeriodicQueueUpdateTime(5*IN_MILLISECONDS), randomBgDifficultyEntry(999, 0, 80, 80, 0)
```
Comment (ShinDarth) [inline src/game/Battlegrounds/BattlegroundMgr.cpp]: please fix indentation

File: src/game/Battlegrounds/BattlegroundMgr.cpp [hunk 6]
```diff
@@ -123,12 +123,16 @@ void BattlegroundMgr::Update(uint32 diff)
...
+        if (time(NULL) > m_NextAutoDistributionTime)
+            {
+                sArenaTeamMgr->DistributeArenaPoints();
+                m_NextAutoDistributionTime = m_NextAutoDistributionTime + BATTLEGROUND_ARENA_POINT_DISTRIBUTION_DAY * sWorld->getIntConfig(CONFIG_ARENA_AUTO_DISTRIBUTE_INTERVAL_DAYS);
+                sWorld->setWorldState(WS_ARENA_DISTRIBUTION_TIME, uint64(m_NextAutoDistributionTime));
+            }
+            m_AutoDistributionTimeChecker = 600000; // check 10 minutes
+         }
+        else
+            m_AutoDistributionTimeChecker -= diff;
```
Comment (ShinDarth) [inline src/game/Battlegrounds/BattlegroundMgr.cpp]: please fix indentation

File: src/game/Battlegrounds/BattlegroundMgr.cpp [hunk 7]
```diff
@@ -684,9 +688,17 @@ void BattlegroundMgr::InitAutomaticArenaPointDistribution()
...
+    time_t curtime = time(NULL);
+    sLog->outDebug(LOG_FILTER_BATTLEGROUND, "TSCR: bg.battleground", "Initializing Automatic Arena Point Distribution");
+    if (wstime < curtime)
+    {
+        m_NextAutoDistributionTime = curtime;           // reset will be called in the next update
+        sLog->outDebug(LOG_FILTER_BATTLEGROUND, "TSCR: bg.battleground", "Battleground: Next arena point distribution time in the past, reseting it now.");
+    }
+    else
+        m_NextAutoDistributionTime = wstime;
+		sLog->outDebug(LOG_FILTER_BATTLEGROUND, "TSCR: bg.battleground", "Automatic Arena Point Distribution initialized.");
```
Comment (ShinDarth) [inline src/game/Battlegrounds/BattlegroundMgr.cpp]: please fix indentation

File: src/game/Battlegrounds/BattlegroundMgr.h [hunk 8]
```diff
@@ -145,6 +147,7 @@ class BattlegroundMgr
         uint32 m_lastClientVisibleInstanceId;
         uint32 m_NextPeriodicQueueUpdateTime;
         time_t m_NextAutoDistributionTime;
+		uint32 m_AutoDistributionTimeChecker;
```
Comment (ShinDarth) [inline src/game/Battlegrounds/BattlegroundMgr.h]: please fix indentation

**Comment by Viste** [comment]
plz close it shit

**Pull request closed without merging**